packages/dids/src/did.methods/did.key.manager.ts

```diff
@@ -0,0 +1,48 @@
+import { driver } from 'did-method-key';
+import { IDidManager } from '../interface/i.did.manager';
+import {Wallet} from '@docknetwork/wallet-sdk-core/lib/modules/wallet';
+
+export class DidKeyManager  implements IDidManager{
+  private static instance: IDidManager;
+
+
+  private wallet;
+
+  private constructor() {
+    this.wallet = Wallet.getInstance();
+  }
+  async createDID(options = {}): Promise<string> {
+    const didDocument = await driver().generate();
+    return didDocument.id.toString()
+  }
+
+  async resolveDID(did: string): Promise<any> {
+    return  driver().get({did});
+  }
+
+  public static getInstance(): IDidManager {
+    if (!DidKeyManager.instance) {
+      DidKeyManager.instance = new DidKeyManager();
+    }
+
+    return DidKeyManager.instance;
+  }
+
+
+  async saveDID(did: string): Promise<boolean> {
```

Comment:
you need to save the whole DID JSON-LD document, not just a string. you also need to save the associated keypairs

Reply:
could make sense if we only used the did:key method, but to support more in the future makes sense to just store the documents so we dont have to ping the chain every time. it will also save the need to have to do expensive crypto operations on the app side through RPC just to view DID info

Reply:
please note though that you should not store the private keys in the document. this library generates keys and exposes the PKs inside the DID document when using generate. we will be sharing these DID documents between agents (wallets) at some point in the future, so we cant expose PK info. thats why they should be stored like account keys are currently in a separate document with an ID that can be referenced